scipy/interpolate/_bsplines.py

```diff
@@ -561,6 +561,87 @@ def integrate(self, a, b, extrapolate=None):
         integral *= sign
         return integral.reshape(ca.shape[1:])
 
+    def design_matrix(self, x, kind="dense", check_finite=True):
```

Comment:
I think this shoud  be a classmethod instead. Otherwise, the usage is a little off (like you do in the tests): first your create a bspline object with (some) coefficients and only then construct the design matrix. Since the design matrix contains basis elements and they are completely defined by the knot vector, the coefficients are superfluous here.
I think the proper usage is more like
```
matr = BSpline.design_matrix(t, x) 
```